plugins/steelseries/fu-steelseries-fizz-gen1.c

```diff
@@ -0,0 +1,281 @@
+#define STEELSERIES_FIZZ_MOUSE_FILESYSTEM_MAIN_APP_ID		0x07U
+#define STEELSERIES_FIZZ_MOUSE_FILESYSTEM_BACKUP_APP_ID		0x08U
+#define STEELSERIES_FIZZ_MOUSE_FILESYSTEM_MSB_DATA_ID		0x09U
+#define STEELSERIES_FIZZ_MOUSE_FILESYSTEM_FACTORY_SETTINGS_ID	0x0aU
+#define STEELSERIES_FIZZ_MOUSE_FILESYSTEM_FSDATA_FILE_ID	0x0bU
+#define STEELSERIES_FIZZ_MOUSE_FILESYSTEM_MAIN_BOOT_ID		0x0cU
+#define STEELSERIES_FIZZ_MOUSE_FILESYSTEM_RECOVERY_ID		0x0eU
+#define STEELSERIES_FIZZ_MOUSE_FILESYSTEM_PROFILES_LIGHTING_ID	0x0fU
+#define STEELSERIES_FIZZ_MOUSE_FILESYSTEM_PROFILES_DEVICE_ID	0x10U
+#define STEELSERIES_FIZZ_MOUSE_FILESYSTEM_FDS_PAGES_ID		0x12U
+#define STEELSERIES_FIZZ_MOUSE_FILESYSTEM_PROFILES_BLUETOOTH_ID 0x13U
+#define STEELSERIES_FIZZ_MOUSE_FILESYSTEM_FREE_SPACE_ID		0xf0U
+
+#define STEELSERIES_FIZZ_PAIRED_STATUS_COMMAND	      0xBBU
+#define STEELSERIES_FIZZ_PAIRED_STATUS_COMMAND_OFFSET 0x00U
+#define STEELSERIES_FIZZ_PAIRED_STATUS_STATUS_OFFSET  0x01U
+
+#define STEELSERIES_FIZZ_CONNECTION_STATUS_COMMAND	  0xBCU
+#define STEELSERIES_FIZZ_CONNECTION_STATUS_COMMAND_OFFSET 0x00U
+#define STEELSERIES_FIZZ_CONNECTION_STATUS_STATUS_OFFSET  0x01U
+
+#define STEELSERIES_FIZZ_BATTERY_LEVEL_COMMAND	      0x92U
+#define STEELSERIES_FIZZ_BATTERY_LEVEL_COMMAND_OFFSET 0x00U
+#define STEELSERIES_FIZZ_BATTERY_LEVEL_LEVEL_OFFSET   0x01U
+
+struct _FuSteelseriesFizzGen1 {
+	FuSteelseriesDevice parent_instance;
+};
+
+static void
+fu_steelseries_fizz_gen1_impl_iface_init(FuSteelseriesFizzImplInterface *iface);
+
+G_DEFINE_TYPE_WITH_CODE(FuSteelseriesFizzGen1,
+			fu_steelseries_fizz_gen1,
+			FU_TYPE_STEELSERIES_DEVICE,
+			G_IMPLEMENT_INTERFACE(FU_TYPE_STEELSERIES_FIZZ_IMPL,
+					      fu_steelseries_fizz_gen1_impl_iface_init))
+
+static gboolean
+fu_steelseries_fizz_gen1_cmd(FuSteelseriesFizzImpl *self,
+			     guint8 *data,
+			     gsize datasz,
+			     gboolean answer,
+			     GError **error)
+{
+	return fu_steelseries_device_cmd(FU_STEELSERIES_DEVICE(self), data, datasz, answer, error);
+}
+
+static gchar *
+fu_steelseries_fizz_gen1_get_version(FuSteelseriesFizzImpl *self, gboolean tunnel, GError **error)
+{
+	guint8 data[STEELSERIES_BUFFER_CONTROL_SIZE] = {0};
+	guint8 cmd = STEELSERIES_FIZZ_VERSION_COMMAND;
+	const guint8 mode = 0U; /* string */
+
+	if (tunnel)
+		cmd |= STEELSERIES_FIZZ_COMMAND_TUNNEL_BIT;
+
+	if (!fu_memwrite_uint8_safe(data,
+				    sizeof(data),
+				    STEELSERIES_FIZZ_VERSION_COMMAND_OFFSET,
+				    cmd,
+				    error))
+		return NULL;
+
+	if (!fu_memwrite_uint8_safe(data,
+				    sizeof(data),
+				    STEELSERIES_FIZZ_VERSION_MODE_OFFSET,
+				    mode,
+				    error))
+		return NULL;
+
+	fu_dump_raw(G_LOG_DOMAIN, "Version", data, sizeof(data));
+	if (!fu_steelseries_fizz_gen1_cmd(self, data, sizeof(data), TRUE, error))
+		return NULL;
+	fu_dump_raw(G_LOG_DOMAIN, "Version", data, sizeof(data));
+
+	fu_device_set_version_format(FU_DEVICE(self), FWUPD_VERSION_FORMAT_TRIPLET);
```

Comment:
can the version format be set in _init?

Reply:
yes, but clarifying the version format with vendor.
Something strange for raw string format read from the devices:
- dongle: 0.01.006.000, but F/W file is 1.0.6
- headset: 000000001.07, but F/W file is 1.0.7

that's why I set it __temporary__ here. I feel the triplet format should be used everywhere.